### ISSUE TITLE
Improve how screen readers announce dates in HTML annotations export

### ISSUE THREAD

**Issue by acelaya** [hypothesis/client#6083]
We are currently using `time` elements to render the various dates included in the HTML report.

These elements have an ISO date in the `datetime` attribute, and their value uses the `YYYY-MM-DD hh:mm` format.

Some screen readers, like VoiceOver and Orca, do not announce them as dates, and instead they announce a regular sequence of text with dashes (`-`) and colons (`:`).

On the other hand, NVDA announces this properly.
